lxd/storage_volumes.go

```diff
@@ -1132,6 +1133,148 @@ func storagePoolVolumePost(d *Daemon, r *http.Request) response.Response {
 
 	r.Body = shared.BytesReadCloser{Buf: &buf}
 
+	target := queryParam(r, "target")
+
+	// Check if clustered.
+	clustered, err := lxdCluster.Enabled(s.DB.Node)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("Failed checking cluster state: %w", err))
+	}
+
+	if clustered && target != "" && req.Source.Location != "" && req.Migration {
+		var sourceNodeOffline bool
+
+		err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+			// Load source node.
+			nodeInfo, err := tx.GetNodeByName(ctx, req.Source.Location)
+			if err != nil {
+				return err
+			}
+
+			sourceAddress := nodeInfo.Address
+
+			if sourceAddress == "" {
+				// Local node.
+				sourceNodeOffline = false
+				return nil
+			}
+
+			sourceMemberInfo, err := tx.GetNodeByAddress(ctx, sourceAddress)
+			if err != nil {
+				return fmt.Errorf("Failed to get source member for %q: %w", sourceAddress, err)
+			}
+
+			sourceNodeOffline = sourceMemberInfo.IsOffline(s.GlobalConfig.OfflineThreshold())
+
+			return nil
+		})
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		var targetProject *api.Project
+		var targetMemberInfo *db.NodeInfo
+
+		if sourceNodeOffline {
+			resp := forwardedResponseIfTargetIsRemote(s, r)
+			if resp != nil {
+				return resp
+			}
+
+			srcPool, err := storagePools.LoadByName(s, srcPoolName)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			if srcPool.Driver().Info().Name == "ceph" {
+				// Load source volume.
+				srcPoolID, err := s.DB.Cluster.GetStoragePoolID(srcPoolName)
+				if err != nil {
+					return response.SmartError(err)
+				}
+
+				var dbVolume *db.StorageVolume
+
+				err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+					dbVolume, err = tx.GetStoragePoolVolume(ctx, srcPoolID, projectName, db.StoragePoolVolumeTypeCustom, volumeName, true)
+					return err
+				})
+				if err != nil {
+					// Check if the user provided an incorrect target query parameter and return a helpful error message.
+					_, volumeNotFound := api.StatusErrorMatch(err, http.StatusNotFound)
+					targetIsSet := r.URL.Query().Get("target") != ""
+					serverIsClustered, _ := lxdCluster.Enabled(s.DB.Node)
+
+					if serverIsClustered && targetIsSet && volumeNotFound {
+						return response.NotFound(fmt.Errorf("Storage volume not found on this cluster member"))
```

Comment:
Im a bit confused by this error as its only for ceph volumes that are not member specific?